spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java

```diff
@@ -111,6 +113,37 @@ private void createPythonScript() {
 
   @Override
   public void open() {
+    // try IPySparkInterpreter first
+    iPySparkInterpreter = getIPySparkInterpreter();
+    if (property.getProperty("zeppelin.spark.useIPython", "true").equals("true") &&
+        iPySparkInterpreter.checkIPythonPrerequisite()) {
+      try {
+        iPySparkInterpreter.open();
+        if (property.getProperty("zeppelin.spark.useIPython", "true").equals("true")) {
+          // don't print it when it is in testing, just for easy output check in test.
+          InterpreterContext.get().out.write(("IPython is available, " +
+              "use IPython for PySparkInterpreter\n")
+              .getBytes());
+        }
+        LOGGER.info("Use IPySparkInterpreter to replace PySparkInterpreter");
+        return;
+      } catch (Exception e) {
+        LOGGER.warn("Fail to open IPySparkInterpreter", e);
+      }
+    }
+    iPySparkInterpreter = null;
+
+    if (property.getProperty("zeppelin.spark.useIPython", "true").equals("true")) {
+      // don't print it when it is in testing, just for easy output check in test.
+      try {
+        InterpreterContext.get().out.write(("IPython is not available, " +
+            "use the native PySparkInterpreter\n")
+            .getBytes());
+      } catch (IOException e) {
+        LOGGER.warn("Fail to write InterpreterOutput", e.getMessage());
```

Comment:
`LOGGER.warn("Fail to write InterpreterOutput", e);`

Reply:
Fixed